services/compute/stopper/src/compute-stop.js

```diff
@@ -0,0 +1,62 @@
+import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+
+export const handler = async (event, context, callback) => {
+  // Extract env vars
+  const { REGION } = process.env;
+
+  if (!REGION) {
+    return callback(new Error('REGION environment variables is required!'));
+  }
+
+  // clients
+  const ec2 = new AWS.EC2({ region: REGION });
+
+  const filters = {
+    DryRun: false,
+    Filters: [
+      {
+        Name: 'instance-state-name',
+        Values: ['running'],
+      },
+      {
+        Name: 'tag:automatedStop',
+        Values: ['true'],
+      },
+    ],
+  };
+
+  try {
+    const instances = [];
+    const response = await ec2.describeInstances(filters).promise();
+
+    response.Reservations.forEach(reservation =>
+      reservation.Instances.forEach(instance => {
+        // If not terminated
+        if (instance.State.Code !== 48) {
+          instances.push(instance.InstanceId);
+        }
+      })
+    );
+
+    if (instances.length) {
+      console.log(`Stopping instances: ${instances.join(', ')}`);
+    }
+
+    const params = {
+      InstanceIds: instances,
+      DryRun: false,
+      Force: false,
+    };
+
+    try {
```

Comment:
is it really necessary to create this nested try/catch block?

Reply:
Not really necessary, the requests could be written as promises with catch.
Regarding environments, only those with `automatedStop` tags are shut down.
`describeInstances`'s [filters](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EC2.html#describeInstances-property) are very flexible though, we can update the filter parameter any time.